libethereum/Executive.cpp

```diff
@@ -369,21 +390,19 @@ bool Executive::executeCreate(Address const& _sender, u256 const& _endowment, u2
 
 OnOpFunc Executive::simpleTrace()
 {
-    return [](uint64_t steps, uint64_t PC, Instruction inst, bigint newMemSize, bigint gasCost, bigint gas, VM* voidVM, ExtVMFace const* voidExt)
-    {
+    Logger& traceLogger = m_vmTraceLogger;
+
+    return [&traceLogger](uint64_t steps, uint64_t PC, Instruction inst, bigint newMemSize,
+               bigint gasCost, bigint gas, VM* voidVM, ExtVMFace const* voidExt) {
         ExtVM const& ext = *static_cast<ExtVM const*>(voidExt);
         VM& vm = *voidVM;
 
-        ostringstream o;
-        o << endl << "    STACK" << endl;
-        for (auto i: vm.stack())
-            o << (h256)i << endl;
-        o << "    MEMORY" << endl << ((vm.memory().size() > 1000) ? " mem size greater than 1000 bytes " : memDump(vm.memory()));
-        o << "    STORAGE" << endl;
-        for (auto const& i: ext.state().storage(ext.myAddress))
-            o << showbase << hex << i.second.first << ": " << i.second.second << endl;
-        dev::LogOutputStream<VMTraceChannel, false>() << o.str();
-        dev::LogOutputStream<VMTraceChannel, false>() << " < " << dec << ext.depth << " : " << ext.myAddress << " : #" << steps << " : " << hex << setw(4) << setfill('0') << PC << " : " << instructionInfo(inst).name << " : " << dec << gas << " : -" << dec << gasCost << " : " << newMemSize << "x32" << " >";
+        BOOST_LOG(traceLogger) << dumpStackMemoryStorage(ext, vm);
```

Comment:
Can we shorten this to `LOG(trace)`?

Reply:
Shortening `BOOST_LOG` to `LOG` sure, but logger name... loggers might be declared in various scopes (and therefore have different or no prefixes).

In general I think recommendation is have a logger defined in a scope of a task performed, usually it's a member of a class doing some job, like `m_execLogger`

Some loggers may be global, like the ones I defined at the bottom in Log.h, then the expression to log is like
```
BOOST_LOG(g_traceLogger::get()) << ...;
```

And here we have a local alias for a member logger (to capture it in lambda)

Reply:
I'm actually not sure if we are going to need more global loggers. Maybe it might be needed if we want to conveniently log into the same channel from different places... But then we could also defined several local loggers writing into the same channel

So yeah, maybe it could be a good idea to 
```
#define LOG(x) BOOST_LOG(m_##x##Logger)
```
and then have a convention to name loggers with `Logger` suffix

Reply:
It might be more convenient to have a logger per module (1-4 files) not per class.

Also, we should probably have a global loggers for errors and warnings, right? Or it should be handled by module loggers with high severity?

Reply:
Logger is just an object used to write logs, and there's a concept of "channels" which represent log topics. And there can be for example several loggers writing to the same channel, or one logger writing to various channels etc.

So I think it will be usually more convenient to have a logger as a member inside one class, even if several classes write to the same channel
Sharing a logger is like sharing other data - either it's global (and might need thread-safety and proper fisco-free initialization), or there's a hassle of passing it around)
Maybe having a global logger inside unnamed namespace in cpp is fine. It's all flexible and we can try different things.

Reply:
As for errors and warnings - I think it's simpler to start with each channel having the same severity in all messages (like it's now) and separate channels for errors and warnings.
There are already `global cwarn`, `cnote`, `cdebug`, `clog` defined in `Log.h`, probably we'll remove some of these and add `cerror`